### PR TITLE
adding support for azure.appconfiguration.v0

### PR DESCRIPTION
**Pull request by vhvb1989**
fix: https://github.com/Azure/azure-dev/issues/3064

used manual generated manifest `apphost-manifest.json`:

```json
{
  "resources": {
    "appconfig": {
      "type": "azure.appconfiguration.v0"
    },
    "webapp": {
      "type": "project.v0",
      "path": "../AspireYouTubeSummariser.WebApp/AspireYouTubeSummariser.WebApp.csproj",
      "env": {
        "OTEL_DOTNET_EXPERIMENTAL_OTLP_EMIT_EXCEPTION_LOG_ATTRIBUTES": "true",
        "OTEL_DOTNET_EXPERIMENTAL_OTLP_EMIT_EVENT_LOG_ATTRIBUTES": "true",
        "ConnectionStrings__appconfig": "{appconfig.connectionString}"
      },
      "bindings": {
        "http": {
          "scheme": "http",
          "protocol": "tcp",
          "transport": "http"
        },
        "https": {
          "scheme": "https",
          "protocol": "tcp",
          "transport": "http"
        }
      }
    }
  }
}
```

### CODE REVIEW

**Review by ellismg — APPROVED** [review]
Small nit around the templating but otherwise looks great.  Thanks, @vhvb1989.

File: cli/azd/resources/apphost/templates/resources.bicept [hunk 1]
```diff
@@ -237,6 +236,25 @@ resource {{bicepName $name}}RoleAssignment 'Microsoft.Authorization/roleAssignme
     roleDefinitionId:  subscriptionResourceId('Microsoft.Authorization/roleDefinitions', '00482a5a-887f-4fb3-b363-3b7fe8e74483')
   }
 }
+{{- end -}}
```
Comment (egostowski) [inline cli/azd/resources/apphost/templates/resources.bicept:239]: thanks

[the remark this answers is not in the document]
Comment (egostowski) [reply cli/azd/resources/apphost/templates/resources.bicept:239]: >

**Review by egostowski — CHANGES_REQUESTED** [review]

**Review by egostowski — APPROVED** [review]

**Comment by ellismg** [comment]
@vhvb1989 - Could you rebase and land this one if you have time?  I think having this support in will be worthwhile - I think we are going to want it either for Preview 3 or early in Preview 4, so if we can land it now that would be great!

**Pull request merged**